src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -272,7 +282,10 @@ private async Task<bool> IsWindowsPlatformNeededAsync()
         private async Task<string> GetTargetFrameworkAliasAsync(string targetFrameworkMoniker)
         {
             IProjectSubscriptionService? subscriptionService = _configuredProject.Services.ProjectSubscription;
-            Assumes.Present(subscriptionService);
+            if (subscriptionService is null)
+            {
+                return string.Empty;
+            }
```

Comment:
If I understand, this is a change in shipping code that exists to support unit tests only?

Or put another way, unit tests won't reach further in this method than here?

In shipping code I feel like the previous assertion was correct. Can you mock the service on the project here, so you don't have to change the product code and can get better test coverage?

Reply:
Thanks for pointing this out. Just for historical reference, we moved away from mocking the service as it was getting wild. Instead, we decided to mock the provider methods that we weren't actually testing here, but using in the way.